subschedule.go

```diff
@@ -0,0 +1,168 @@
+type SubscriptionScheduleStatus string
+
+// List of values that SubscriptionScheduleStatus can take.
+const (
+	SubscriptionScheduleStatusActive    SubscriptionScheduleStatus = "active"
+	SubscriptionScheduleStatusCanceled  SubscriptionScheduleStatus = "canceled"
+	SubscriptionScheduleStatusCompleted SubscriptionScheduleStatus = "completed"
+	SubscriptionScheduleStatusPastDue   SubscriptionScheduleStatus = "not_started"
+	SubscriptionScheduleStatusTrialing  SubscriptionScheduleStatus = "released"
+)
+
+// SubscriptionScheduleRenewalBehavior describe what happens to a schedule when it ends.
+type SubscriptionScheduleRenewalBehavior string
+
+// List of values that SubscriptionScheduleRenewalBehavior can take.
+const (
+	SubscriptionScheduleRenewalBehaviorNone    SubscriptionScheduleRenewalBehavior = "none"
+	SubscriptionScheduleRenewalBehaviorRenew   SubscriptionScheduleRenewalBehavior = "release"
+	SubscriptionScheduleRenewalBehaviorRelease SubscriptionScheduleRenewalBehavior = "renew"
+)
+
+// SubscriptionSchedulePhaseItemParams is a structure representing the parameters allowed to control
+// a specic plan on a phase on a subscription schedule.
+type SubscriptionSchedulePhaseItemParams struct {
+	BillingThresholds *SubscriptionItemBillingThresholdsParams `form:"billing_thresholds"`
+	Plan              *string                                  `form:"plan"`
+	Quantity          *int64                                   `form:"quantity"`
+}
+
+// SubscriptionSchedulePhaseParams is a structure representing the parameters allowed to control
+// a phase on a subscription schedule.
+type SubscriptionSchedulePhaseParams struct {
+	ApplicationFeePercent *int64                                 `form:"application_fee_percent"`
+	Coupon                *string                                `form:"coupon"`
+	EndDate               *int64                                 `form:"end_date"`
+	Iterations            *int64                                 `form:"iterations"`
+	Plans                 []*SubscriptionSchedulePhaseItemParams `form:"plans"`
+	Trial                 *bool                                  `form:"trial"`
+	TrialEnd              *int64                                 `form:"trial_end"`
+}
+
+// SubscriptionScheduleParams is the set of parameters that can be used when creating or updating a
+// subscription schedule.
+type SubscriptionScheduleParams struct {
+	Params            `form:"*"`
+	Billing           *string                              `form:"billing"`
+	BillingThresholds *SubscriptionBillingThresholdsParams `form:"billing_thresholds"`
+	Customer          *string                              `form:"customer"`
+	FromSubscription  *string                              `form:"from_subscription"`
+	DaysUntilDue      *int64                               `form:"days_until_due"`
+	Phases            []*SubscriptionSchedulePhaseParams   `form:"phases"`
+	RenewalBehavior   *string                              `form:"renewal_behavior"`
```

Comment:
Just flagging that the docs I'm looking at don't have `DaysUntilDue`, and they `EndDate`, which seems to be missing here.

Reply:
Damn nice catch. ~daysUntilDue` should be in a separate struct/hash. I fixed this.

Can you clarify what you meant about EndDate? This should only be in the `current_phase` or `phases[]` instead

Reply:
Arg, sorry — I mixed up a word, what I meant is that it looks like there should be a `StartDate`.

All I have is an internal link, but if you scroll to "create a schedule" and look near the bottom, you should see a `start_date` underneath `renewal_behavior`.